### PR TITLE
Retry loading serialized job details

### LINKED ISSUE #45954: [oxygen] msgpack errors in test suite
The following shows on the console in some integration tests:

```
03:55:54   test_multiple_sources_master_dev_merge_lists (integration.pillar.test_git_pillar.TestGitPythonAuthenticatedHTTP)

03:55:55 03:55:54,575 [salt.payload                               :147 ][CRITICAL] Could not deserialize msgpack message. This often happens when trying to read a file not in binary mode. To see message payload, enable debug logging and retry. Exception: Unpack failed: error = 0
03:55:55 03:55:54,686 [tornado.application                        :611 ][ERROR   ] Exception in callback <functools.partial object at 0xbd5edb8>
03:55:55 Traceback (most recent call last):
03:55:55   File "/usr/lib64/python2.7/site-packages/tornado/ioloop.py", line 591, in _run_callback
03:55:55     ret = callback()
03:55:55   File "/usr/lib64/python2.7/site-packages/tornado/stack_context.py", line 274, in null_wrapper
03:55:55     return fn(*args, **kwargs)
03:55:55   File "/testing/salt/minion.py", line 3139, in _process_event
03:55:55     self.mminion.returners[fstr](data['jid'])
03:55:55   File "/testing/salt/returners/local_cache.py", line 300, in get_load
03:55:55     ret = serial.load(rfh)
03:55:55   File "/testing/salt/payload.py", line 166, in load
03:55:55     return self.loads(data)
03:55:55   File "/testing/salt/payload.py", line 137, in loads
03:55:55     ret = msgpack.loads(msg, use_list=True, encoding=encoding)
03:55:55   File "msgpack/_unpacker.pyx", line 210, in msgpack._unpacker.unpackb
03:55:55 UnpackValueError: Unpack failed: error = 0

03:56:08 [CPU:0.0%|MEM:35.7%]  ... OK (14.152s)
```

This is likely due to unicode-related changes causing the payload to no longer be bytes. Should not be hard to track down. *(EDIT: famous last words...)*

### PR DESCRIPTION
**Pull request by terminalmage**
This works around a race condition that can occur in the test suite when the event handler tries to process an event before the job's serialized details have finished being written to disk.

@cachedout I tried to err on the side of caution here, by trying 5 times and waiting a quarter-second between each try. In my testing, just a single retry (i.e. 2 tries total) got rid of all instances of this traceback, but I figured there wasn't any harm in trying to account for longer delays. I'm open to tweaking this if you think it's worth doing.

Resolves #45954.

**NOTE: This traceback does not affect test results, it's in the event handler. To confirm whether or not this fixed the traceback, you will need to check the full console output for the job in Jenkins and look for the string `UnpackValueError`.**

### CODE REVIEW

**Comment by terminalmage** [comment]
No `UnpackValueError` showed up in the console log on either of the PY2 tests. This should be safe to merge, assuming we get a sign-off from @cachedout on it. The only place I remember seeing this though is in the branch builds, which run on a wider array of platforms, and we won't be able to see if it fixes the traceback there until after it's merged.

**Review by rallytime — APPROVED** [review]
This LGTM, and I didn't see any `UnpackValueError`s in any of the logs for the test runs. However, after discussing this with Erik offline, he said he was testing this with Python2 on Ubuntu 16, which PR tests don't run.

I'd like @cachedout to look this over before merging however.

**Comment by cachedout** [comment]
This is a sensitive area of the codebase and we have had problems here before, but I think this is relatively safe.

**Review by cachedout — APPROVED** [review]

**Comment by terminalmage** [comment]
Now that this is merged, I've triggered a jenkins build for Ubuntu 16 PY2. Here's the URL: https://jenkins.saltstack.com/job/oxygen.rc1/job/oxygen-salt-ubuntu16.04-py2/27

Note that at the time of this writing the job hasn't yet started, it's currently awaiting an executor. So until the job actually starts running you'll get a 404 error if you visit that URL.

Remember, the test results are unaffected by the bug this PR aims to fix, so take note of the bold message at the bottom of the [OP](https://github.com/saltstack/salt/pull/45998#issue-168751541) to know how to check whether or not this PR succeeded.

**Comment by rallytime** [comment]
@terminalmage I checked the logs and this looks good.

**Comment by terminalmage** [comment]
Yep, confirmed! :beers:

**Pull request merged**
